pkg/node/manager/manager.go

```diff
@@ -358,6 +358,13 @@ func (m *Manager) NodeUpdated(n nodeTypes.Node) {
 		}
 	}
 
+	var ipsAdded, healthIPsAdded []string
+
+	// helper function with the required logic to skip IPCache interactions
+	skipIPCache := func(address nodeTypes.Address) bool {
+		return m.legacyNodeIpBehavior() && address.Type != addressing.NodeCiliumInternalIP
+	}
```

Comment:
Nit: Why not define this as a normal function outside the scope of this inner function?

Reply:
I thought a little bit about this as well. It didn't make sense to me to have since it is only used inside this function. Like, the logic the function is testing does not make sense elsewhere.